types/framework/entities/actor.d.ts

```diff
@@ -294,47 +294,56 @@ declare class Actor<
   /** @override */
   createEmbeddedEntity(
     embeddedName: string,
-    data: Optional<ActiveEffect | Actor.OwnedItem<I>>,
+    data: Optional<ActiveEffect.Data | Actor.OwnedItem<I>>,
     options?: any
-  ): Promise<ActiveEffect | Actor.OwnedItem<I>>;
+  ): Promise<ActiveEffect.Data | Actor.OwnedItem<I>>;
 
   /**
    * When Owned Items are created process each item and extract Active Effects to transfer to the Actor.
    * @param created - Created owned Item data objects
    * @param temporary - Is this a temporary item creation?
    * @returns An array of effects to transfer to the Actor
    */
-  _createItemActiveEffects(created: ActiveEffect, { temporary }?: { temporary?: boolean }): ActiveEffect;
+  _createItemActiveEffects(
+    created: Actor.OwnedItem<I> | Array<Actor.OwnedItem<I>>,
+    { temporary }?: { temporary?: boolean }
+  ): ActiveEffect.Data[] | ActiveEffect.Data;
```

Comment:
In which cases does this return only a single `ActiveEffect.Data`? Can we specify with generics, depending on the input?

Reply:
It depends on a lot of things: If several `OwnedItem`s have been created, it will return any effects from all those items. An `OwnedItem` also might have more than one effect attached to it, so that also affects it. And finally, only effects that have the `transfer` flag will be created and returned.

If all of this result in only one effect being created, the return type will be `ActiveEffect.Data`. If more than one is created, it is `ActiveEffect.Data[]`. But I just noticed that I missed something: If no effect is created, nothing is returned. We need to reflect that in the return type as well.

Practically speaking, I don't think we can give a more concrete return type as we would need to inspect on the number of effects that have the `transfer` flag set. That does not sound reasonable to me.

Reply:
Yeah in that case, just add the type when nothing is returned. I assume the other functions where I commented are similar?

Reply:
I'll take a closer look at the others. I think we might have more luck with those.